### DISCUSSION TITLE
Display current focus rectangle when dialog is open

### DISCUSSION THREAD

**Discussion by Jakooth** [microsoft/fluentui#26545] [Ideas] [answered]
As simple as this, when you open a dialog the focus rectangle is not displayed on the default active element - one has to use tab to see it. This is a small accessibility concern.
In fact it used to work at some point, when ms-Fabric--isFocusVisible/Hidden was set on the document body.
However within a recent update the focus class is now set on a child div where react is rendering - since dialogs are appended directly to the body the selector pattern: .ms-Fabric--isFocusVisible .root-[n]:focus::after no longer works.
But even when it worked, it was a little of a workaround, because it is assumed you have to use tab key before dialog is open, but for example, we have cases, where alert dialog is immediately displayed.
The idea is to either decouple the focus visible class and use :focus-visible, or the automatically set ms-Fabric--isFocusVisible on the dialog placeholder when opened via keyboard.
Cheers!

**Comment by Jakooth** [comment]
I was digging more into this today and in fact this issue: https://github.com/microsoft/fluentui/issues/24588, could workaround the problem, because we can use setFocusVisibility after dialog is open as long es the current focus provider is automatically determined from context - now it appends classes to the body.

**Comment by micahgodbolt** [comment, marked as answer]
Discussions is not the place to file/discuss bugs.
